plugins/reach/lib/kReachManager.php

```diff
@@ -3,8 +3,166 @@
 /**
  * @package plugins.reach
  */
-class kReachManager implements kObjectChangedEventConsumer, kObjectCreatedEventConsumer, kObjectAddedEventConsumer
+class kReachManager implements kObjectChangedEventConsumer, kObjectCreatedEventConsumer, kObjectAddedEventConsumer, kGenericEventConsumer
 {
+	/**
+	 * @var array<booleanNotificationTemplate>
+	 */
+	protected static $booleanNotificationTemplatesFulfilled;
+	protected static $booleanNotificationTemplatesFromReachProfiles;
+	protected static $reachProfilesFilteredWithoutBooleanEventNotifications;
+
+	static $isInit = false;
+
+	protected function getObjectType($eventObjectClassName)
+	{
+		switch($eventObjectClassName)
+		{
+			case "entry":
+				return  objectType::ENTRY;
+			case "category":
+				return objectType::CATEGORY;
+			case "asset":
+				return objectType::ASSET;
+			case "flavorAsset":
+				return objectType::FLAVORASSET;
+			case "thumbAsset":
+				return objectType::THUMBASSET;
+			case "uiconf":
+				return  objectType::UICONF;
+			case "conversionProfile2":
+				return objectType::CONVERSIONPROFILE2;
+			case "kuser":
+				return objectType::KUSER;
+			case "permission":
+				return objectType::PERMISSION;
+			case "permissionItem":
+				return  objectType::PERMISSIONITEM;
+			case "userRole":
+				return  objectType::USERROLE;
+			default:
+				return null;
+		}
+	}
+	/* (non-PHPdoc)
+	 * @see kGenericEventConsumer::consumeEvent()
+	 */
+	public function consumeEvent(KalturaEvent $event)
+	{
+		$scope = $event->getScope();
```

Comment:
1. This function must return true.
2. Move its content to reachCosnumeEvent and you can use its return value,